packages/hw-app-btc/src/finalizeInput.ts

```diff
@@ -1,37 +1,35 @@
-// @flow
 import Transport from "@ledgerhq/hw-transport";
 import { bip32asBuffer } from "./bip32";
 import { MAX_SCRIPT_BLOCK } from "./constants";
-
 export function provideOutputFullChangePath(
-  transport: Transport<*>,
+  transport: Transport,
   path: string
-): Promise<string> {
-  let buffer = bip32asBuffer(path);
+): Promise<Buffer> {
+  const buffer = bip32asBuffer(path);
   return transport.send(0xe0, 0x4a, 0xff, 0x00, buffer);
 }
-
 export async function hashOutputFull(
-  transport: Transport<*>,
+  transport: Transport,
   outputScript: Buffer,
   additionals: Array<string> = []
 ): Promise<void> {
   let offset = 0;
-  let p1 = 0x80;
+  const p1 = 0x80;
   const isDecred = additionals.includes("decred");
+
   ///WARNING: Decred works only with one call (without chunking)
   //TODO: test without this for Decred
   if (isDecred) {
-    return transport.send(0xe0, 0x4a, p1, 0x00, outputScript);
+    await transport.send(0xe0, 0x4a, p1, 0x00, outputScript);
```

Comment:
⚠️ the logic changed!! we used to return (meaning STOP the logic). whereas here we will not return and continue on the while.